news/_posts/2022-07-01-introducing-horizontal-migration.md

```diff
@@ -0,0 +1,74 @@
+
+Currently, _migrating_ one page to Symfony involves several steps:
+
+- Analyze the page's features and its business logic, then specify any unclear or ambiguous behavior
+- Create a new Symfony controller, identify the different "actions" and set them up (routing, access control...)
+- Reimplement the page's original behavior using modern components (Grid, Symfony forms, Twig, [CQRS](https://devdocs.prestashop.com/1.7/development/architecture/migration-guide/strategy/#phase-3-cqrscqrs)...)
+- Implement any feature required in order to reproduce the original behavior using the new components (Twig layouts, form theme, js...)
+- Implement adapters for any legacy component used in the original controller (as required by the [architecture rules](/releases/new-architecture-1-6-1-0/) established in 2015)
+- Fix old bugs discovered while reimplementing the old features, or reimagine the existing features to improve them
+- Add automated tests
+- Remove the old controller and rerouting it to the new controller
+
+That's quite a lot of steps, right? 
+
+To us, this shows that the _migration_ process isn't really just about changing the underlying framework or a couple components. It's much more than that: **it's a full-on refactoring.**
+
+At the time of writing, over 90 Back office pages have been _refactored_ using the process above. It's a good, thorough process, and we are proud of how it has contributed to increasing PrestaShop's overall software quality during recent years. However, it has its downsides.
+
+Migrating a page this way is slow and requires a considerable effort. It basically amounts to building the page again – a _better_ version of it. With several dozen modified files, refactoring a typical page takes several weeks of development, and months of review and refinement. There are still 50 pages left to migrate in the Back office. That's _years'_ worth of work! 
+
+In the meantime, PrestaShop's architecture has become heterogeneous. Although the pages migrated to Symfony benefit from all the modern concepts built over the years, a third of the Back office pages remain largely identical to how they were in PrestaShop 1.6 – not to mention the whole Front office and the Webservice API. 
+
+This means that developers have to deal with some parts of the software working in a certain way and others in a completely different one. As we have mentioned in previous articles, this is bad because having more than one way of doing the same thing carries a higher risk of bugs, lower performance, more maintenance work, and a steeper learning curve for people who are new to the platform.
+
+Moreover, every bit of the legacy framework must be preserved until the migration has been fully completed. Even if in a few years only a single page that few people use was left to migrate, everything it relies on, Dispatcher, ObjectModel, Smarty, Helpers... the entire legacy stack must remain in place, alive and maintained.
```

Comment:
```suggestion
Moreover, all parts of the legacy framework must be preserved until the migration is fully completed. In a few years, a single page with few users is left to migrate, with many dependencies, such as Dispatcher, ObjectModel, Smarty, Helpers: this page could have an impact on the migration's health. Therefore, the entire legacy stack must remain in place, alive and maintained.
```

Reply:
Hi team, could you validate the technical accuracy of this edit I've just made? Thanks a lot.

Reply:
Nope, you're changing the meaning. My intent is to show that our current strategy forces us to keep and maintain every single part of the legacy system until we reach 100% progress in the migration project. It's binary: even if a few years from now we finish migrating all pages but one, every last component must remain in place because they support that single remaining page. We will only be able to remove these components once that hypothetical final page has been migrated as well, and we reach 100% project completion.

Reply:
Ok fine with me, I'm deleting this change